tasks/installer-setup.yml

```diff
@@ -0,0 +1,158 @@
+  ignore_errors: true
+  when: ansible_facts.os_family in ["RedHat", "Rocky", "AlmaLinux"] and not ansible_check_mode and ansible_pkg_mgr == "dnf"
+
+- name: Install latest datadog-agent package (yum)
+  yum:
+    name: "datadog-installer"
+    update_cache: true
+    state: latest # noqa package-latest
+  register: datadog_installer_install_result
+  ignore_errors: true
+  when: ansible_facts.os_family in ["RedHat", "Rocky", "AlmaLinux"] and not ansible_check_mode and ansible_pkg_mgr == "yum"
+
+- name: Install Datadog installer (apt)
+  apt:
+    name: "datadog-installer"
+    state: latest # noqa package-latest
+    update_cache: true
+    cache_valid_time: "{{ datadog_apt_cache_valid_time }}"
+  register: datadog_installer_install_result
+  ignore_errors: true
+  when: ansible_facts.os_family == "Debian" and not ansible_check_mode
+
+- name: Bootstrap the installer
+  command: /usr/bin/datadog-bootstrap bootstrap
+  register: datadog_installer_bootstrap_result
+  environment:
+    DATADOG_TRACE_ID: "{{ datadog_installer_trace_id }}"
+    DATADOG_PARENT_ID: "{{ datadog_installer_trace_id }}"
+    DD_SITE: "{{ datadog_site | default('datadoghq.com') }}"
+    DD_API_KEY: "{{ datadog_api_key }}"
+    DD_REMOTE_UPDATES: "{{ 'true' if datadog_remote_updates is defined and datadog_remote_updates else '' }}"
+    DD_APM_INSTRUMENTATION_ENABLED: "{{ datadog_apm_instrumentation_enabled }}"
+    DD_APM_INSTRUMENTATION_LANGUAGES: "{{ datadog_apm_instrumentation_libraries | join(',') }}"
+  ignore_errors: true
+  when: not datadog_installer_install_result.failed
+  changed_when: true
+
+- name: Check if installer owns datadog-agent package
+  command: datadog-installer is-installed "{{ datadog_agent_flavor }}"
+  failed_when: datadog_installer_owns_agent.rc != 0 and datadog_installer_owns_agent.rc != 10
+  register: datadog_installer_owns_agent
+  when: not datadog_installer_bootstrap_result.failed
+  changed_when: true
+
+- name: Disable agent install if owned by installer
+  set_fact:
+    agent_datadog_skip_install: true
+  when: datadog_installer_owns_agent.rc == 0
+
+- name: Query APM packages owned by installer
+  command: datadog-installer is-installed "datadog-apm-library-{{ item }}"
+  register: datadog_installer_owned_apm_packages
+  loop: "{{ datadog_apm_instrumentation_libraries }}"
+  when: not datadog_installer_bootstrap_result.failed
+  failed_when: datadog_installer_owned_apm_packages.rc != 0 and datadog_installer_owned_apm_packages.rc != 10
+  changed_when: true
+  environment:
+    DD_API_KEY: "{{ datadog_api_key }}"
+
+- name: Save APM packages before filtering
+  set_fact:
+    datadog_apm_instrumentation_libraries_unfiltered: "{{ datadog_apm_instrumentation_libraries }}"
+
+- name: Filter APM packages owned by the installer
+  set_fact:
+    datadog_apm_instrumentation_libraries: "{{ datadog_apm_instrumentation_libraries | difference([item.item]) }}"
+  when: item.rc == 0
+  loop: "{{ datadog_installer_owned_apm_packages.results }}"
```

Comment:
Does this work when the var isn't registered or should we add `when: not datadog_installer_bootstrap_result.failed`?

Reply:
Let's be on the safer side, I added the clause